content/hardware/06.nicla/boards/nicla-voice/tutorials/ei-intruder-detector/content.md

```diff
@@ -0,0 +1,474 @@
+
+const char SSID[]     = "xxxxxxxx";    // Network SSID (name)
+const char PASS[]     = "xxxxxxxx";    // Network password (use for WPA, or use as key for WEP)
+
+String DoorEvent = "System initiated";  
+int battery;
+bool BLEstatus;
+bool alertStatus;
+
+void initProperties(){
+
+  ArduinoCloud.setThingId(THING_ID);
+  ArduinoCloud.addProperty(DoorEvent, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(battery, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(BLEstatus, READ, ON_CHANGE, NULL);
+  ArduinoCloud.addProperty(alertStatus, READ, ON_CHANGE, NULL);
+
+}
+
+WiFiConnectionHandler ArduinoIoTPreferredConnection(SSID, PASS);
+```
+The main responsibility of the Portenta H7 code is to connect to the Nicla Voice and be aware of new notifications to forward them to the Cloud. To achieve this, the first thing done by the Portenta is to initialize the BLE communication, then initialize the Arduino IoT Cloud service and start scanning for peripherals.
+
+```arduino
+  // Initialize BLE
+  if (!BLE.begin()) {
+    Serial.println("Starting BLE failed!");
+    while (1) {
+    }
+  }
+
+  // Defined in thingProperties.h
+  initProperties();
+
+  // Connect to Arduino IoT Cloud
+  if (!ArduinoCloud.begin(ArduinoIoTPreferredConnection)) {
+    Serial.println("ArduinoCloud.begin FAILED!");
+  }
+
+  setDebugMessageLevel(2);
+  ArduinoCloud.printDebugInfo();
+
+    // start scanning for peripheral
+  BLE.scan();
+  Serial.println("Scanning for peripherals.");
+```
+The scanning process will automatically stop when it finds a "Nicla Lock" called device, then it will search for its services and characteristics with the `NiclaLockHandler()` function.
+
+```arduino
+
+void loop() {
+
+  // check if a peripheral has been discovered
+  BLEDevice peripheral = BLE.available();
+
+  // Turn on the Green LED if the board is successfully connected to the Cloud. On with Low.
+  if (ArduinoCloud.connected()) {
+    digitalWrite(LEDG, LOW);
+  } else {
+    digitalWrite(LEDG, HIGH);
+  }
+
+  if (peripheral) {
+    // peripheral discovered, print out address, local name, and advertised service
+    Serial.print("Found ");
+    Serial.print(peripheral.address());
+    Serial.print(" '");
+    Serial.print(peripheral.localName());
+    Serial.print("' ");
+    Serial.print(peripheral.advertisedServiceUuid());
+    Serial.println();
+
+    // Check if the peripheral is a Nicla Lock:
+    if (peripheral.localName() == "Nicla Lock") {
+      // stop scanning
+      BLE.stopScan();
+
+      // Nicla Voice node connection handler
+      NiclaLockHandler(peripheral);
+
+      // peripheral disconnected, start scanning again
+      BLE.scan();
+    }
+  }
+  ArduinoCloud.update();
+}
+
+```
+
+In order to just get connected with the Nicla Voice, the Portenta searches for the specific "Alert Service" using its UUID `1802` and for the necessary characteristics, the "Battery Level" using the `2A19` UUID  and the "Alert Level" using the `2A06` UUID.
```

Comment:
@TaddyHC or @canchebagur any idea why the render engine do these strange things with the numbers¿?

![image](https://user-images.githubusercontent.com/7687624/231131053-a05bcfb8-d700-4375-90bc-e02bea874e5c.png)